wire-compiler/src/main/java/com/squareup/wire/java/RxJavaServiceFactory.java

```diff
@@ -0,0 +1,125 @@
+
+    TypeSpec.Builder typeBuilder = TypeSpec.classBuilder(interfaceName.simpleName());
+    typeBuilder.addModifiers(Modifier.PUBLIC, FINAL);
+
+    if (!service.documentation().isEmpty()) {
+      typeBuilder.addJavadoc("$L\n", service.documentation());
+    }
+
+    TypeSpec.Builder endpointBuilder = TypeSpec.interfaceBuilder(endpointName.simpleName());
+    endpointBuilder.addModifiers(Modifier.PUBLIC, Modifier.STATIC);
+
+    typeBuilder.addMethod(MethodSpec.constructorBuilder()
+        .addAnnotation(INJECT)
+        .addModifiers(PUBLIC)
+        .addParameter(endpointName, "endpoint")
+        .addStatement("this.endpoint = endpoint")
+        .build());
+
+    for (WireRpc rpc : service.rpcs()) {
+      ProtoTypeName requestType = rpc.requestType();
+      TypeName requestJavaType = javaGenerator.typeName(requestType);
+      ProtoTypeName responseType = rpc.responseType();
+      TypeName responseJavaType = javaGenerator.typeName(responseType);
+
+      String methodName = upperToLowerCamel(rpc.name());
+      MethodSpec.Builder rpcBuilder = MethodSpec.methodBuilder(methodName);
+      rpcBuilder.addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT);
+      rpcBuilder.returns(responseJavaType);
+      rpcBuilder.addAnnotation(AnnotationSpec.builder(POST)
+          .addMember("value", "$S", "/" + service.protoTypeName() + "/" + rpc.name())
+          .build());
+
+      rpcBuilder.addParameter(ParameterSpec.builder(requestJavaType, "request")
+          .addAnnotation(BODY)
+          .build());
+
+      if (!rpc.documentation().isEmpty()) {
+        rpcBuilder.addJavadoc("$L\n", rpc.documentation());
+      }
+
+      endpointBuilder.addMethod(rpcBuilder.build());
+
+      TypeName functionType = ParameterizedTypeName.get(FUNC1, requestJavaType, responseJavaType);
+      typeBuilder.addField(FieldSpec.builder(functionType, methodName)
+          .addModifiers(PRIVATE, FINAL)
+          .initializer("$L", TypeSpec.anonymousClassBuilder("")
+              .addSuperinterface(functionType)
+              .addMethod(MethodSpec.methodBuilder("call")
+                  .addAnnotation(Override.class)
+                  .addModifiers(PUBLIC)
+                  .returns(responseJavaType)
+                  .addParameter(requestJavaType, "request")
+                  .addStatement("return endpoint.$L(request)", methodName)
+                  .build())
+              .build())
+          .build());
+
+      typeBuilder.addMethod(MethodSpec.methodBuilder("get" + rpc.name())
+          .addModifiers(PUBLIC)
+          .returns(functionType)
+          .addStatement("return $L", methodName)
+          .build());
+    }
+
+    typeBuilder.addField(endpointName, "endpoint", PRIVATE, FINAL);
+    typeBuilder.addType(endpointBuilder.build());
+    return typeBuilder.build();
+  }
+
+  private String upperToLowerCamel(String string) {
+    return UPPER_CAMEL.to(LOWER_CAMEL, string);
+  }
+
+  protected ClassName interfaceName(
```

Comment:
I don't think this really makes sense as a general purpose extension point, but it gets us closer to the goal of the switch. Once that happens a lot of clean up should be possible.